core/src/main/java/com/digitalpebble/stormcrawler/protocol/okhttp/HttpProtocol.java

```diff
@@ -570,12 +571,20 @@ public Response intercept(Interceptor.Chain chain) throws IOException {
             byte[] encodedBytesRequest =
                     Base64.getEncoder().encode(requestverbatim.toString().getBytes());
 
+            StringBuilder protocols = new StringBuilder(response.protocol().toString());
```

Comment:
mark as _final_? same for _handshake_?
Doing so makes things a bit more explicit, possibly a bit faster but more importantly can highlight unexpected reassignments.